ssz/simple-serialize.md

```diff
@@ -170,14 +170,26 @@ fixed_parts = [part if part != None else variable_offsets[i] for i, part in enum
 return b"".join(fixed_parts + variable_parts)
 ```
 
-If `value` is a union type:
+### Union
 
-Define value as an object that has properties `value.value` with the contained value, and `value.type_index` which indexes the type.
+A `value` as `Union[T...]` type has properties `value.value` with the contained value, and `value.selector` which indexes the selected `Union` type option `T`.
+
+A `Union`:
+- May have multiple selectors with the same type. 
+- Should not use selectors above 127 (i.e. highest bit is set), these are reserved for backwards compatible extensions.
+- Must have at least 1 type option.
+- May have `None` as first type option, i.e. `selector == 0`
+- Must have at least 2 type options if the first is `None`
+- Is always considered a variable-length type, even if all type options have an equal fixed-length.
 
 ```python
-serialized_bytes = serialize(value.value)
-serialized_type_index = value.type_index.to_bytes(BYTES_PER_LENGTH_OFFSET, "little")
-return serialized_type_index + serialized_bytes
+if value.value is None:
+    assert value.selector == 0
+    return b"\x00"
+else:
+    serialized_bytes = serialize(value.value)
+    serialized_selector_index = value.selector.to_bytes(1, "little")
+    return serialized_selector_index + serialized_bytes
```

Comment:
(nit) may be define `serialize(None) = Bytes()` and leave just `else` block:
```suggestion
serialized_bytes = serialize(value.value)
serialized_selector_index = value.selector.to_bytes(1, "little")
return serialized_selector_index + serialized_bytes
```

Reply:
When the type is `None`, the selector `0` value still needs to be encoded, hence the `b"\x00"`, I think a little verbosity is fine here.